src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -36,74 +38,106 @@ const propTypes = {
+                    errors[`beneficialOwner.${ownerID}.state`] = this.props.translate('bankAccount.error.addressState');
+                }
+
+                if (!ValidationUtils.isRequiredFulfilled(values[`beneficialOwner.${ownerID}.zipCode`])
+                    || !ValidationUtils.isValidZipCode(values[`beneficialOwner.${ownerID}.zipCode`])) {
+                    errors[`beneficialOwner.${ownerID}.zipCode`] = this.props.translate('bankAccount.error.zipCode');
+                }
+            });
         }
 
-        const errors = {};
-        _.each(this.requiredFields, (inputKey) => {
-            if (ValidationUtils.isRequiredFulfilled(this.state[inputKey])) {
-                return;
-            }
+        if (!ValidationUtils.isRequiredFulfilled(values.acceptTermsAndConditions)) {
+            errors.acceptTermsAndConditions = this.props.translate('common.error.acceptedTerms');
+        }
 
-            errors[inputKey] = true;
-        });
-        BankAccounts.setBankAccountFormValidationErrors({...errors, beneficialOwnersErrors});
-        return _.every(beneficialOwnersErrors, _.isEmpty) && _.isEmpty(errors);
+        if (!ValidationUtils.isRequiredFulfilled(values.certifyTrueInformation)) {
+            errors.certifyTrueInformation = this.props.translate('beneficialOwnersStep.error.certify');
+        }
+
+        return errors;
     }
 
-    removeBeneficialOwner(beneficialOwner) {
+    removeBeneficialOwner(ownerID) {
         this.setState((prevState) => {
-            const beneficialOwners = _.without(prevState.beneficialOwners, beneficialOwner);
+            const beneficialOwners = _.without(prevState.beneficialOwners, ownerID);
 
             // We set 'beneficialOwners' to null first because we don't have a way yet to replace a specific property without merging it.
             // We don't use the debounced function because we want to make both function calls.
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners: null});
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
+            FormActions.setDraftValues(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {beneficialOwners: null});
+            FormActions.setDraftValues(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {beneficialOwners});
 
-            // Clear errors
-            BankAccounts.setBankAccountFormValidationErrors({});
             return {beneficialOwners};
         });
     }
 
     addBeneficialOwner() {
-        this.setState(prevState => ({beneficialOwners: [...prevState.beneficialOwners, {}]}));
+        this.setState((prevState) => {
+            const beneficialOwners = [...prevState.beneficialOwners, NumberUtils.rand64()];
+
+            // We set 'beneficialOwners' to null first because we don't have a way yet to replace a specific property without merging it.
+            // We don't use the debounced function because we want to make both function calls.
```

Comment:
I feel like this is not true anymore. Our updated array merging behavior in Onyx should be that when we change an array value for an object property the entire array is replaced.

Reply:
That's good to know! Can I just use the second merge in that case?

Reply:
I think so - but I'd also test it out to be sure.

Reply:
>Our updated array merging behavior in Onyx should be that when we change an array value for an object property the entire array is replaced.

Nice! TIL!